### ISSUE TITLE
Audio Lag in Rhythm Tengoku

### ISSUE THREAD

**Issue by internetakias** [p1pkin/demul#361]
Hey, I'm using the latest demul alpha on Windows 7 64 bit and I'm experiencing audio lag with Rhythm Tengoku, and with it being a rhythm game that relies on sound and precise inputs it's kinda hard to play.
I've tried lowering the sound buffer size but anything below 2048 lowers the framerate for me.
Any way this could be fixed?

**Comment by Nashismo** [comment]
Mate I was suffering audio lag on all my games with Demul AND I found a fix!! 

First let me tell you that I am no noob user, I am 35 and I know windows in and out, there was no configuration I didn't change and nothing worked to make this emu work correctly.

Now for the fix: For some strange reason if I use SMAA injector tool in Demul my games all work without sound distortion and even perform better than before!
It is a problem with the GPU or CPU that is not working hard enough with Demul and the way I see it, this SMAA injector forces the GPU/CPU to work as they should in the first place.

I am running:
Windows 7 64 bits, latest Demul
i5 2400 3.3GHZ, 8GB RAM, Nvidia 660ti.

You can download the SMAA tool at
http://mrhaandi.blogspot.cl/p/injectsmaa.html

Actually this SMAA actually works for me, as the FXAA within the emu never worked for me. I hope the creator of the Emu could somehow use this info because ALL MY GAMES work great now (and with good Antialising).

**Comment by p1pkin** [comment]
@Nashismo you are messing with terms, term 'lag' actually means 'it have delay' however it sounds with no glitches

in your case it is speed drops (and as result bad sound) because of bad Nvidia drivers power management. there no need to use 3rd-party tools, it can be disabled in settings:
Nvidia Control Panel > Manage 3D Settings > Program Settings > Select Demul.exe > Scroll down to power management and select "Prefer Maximum Performance"

btw its not something Demul-specific, same crap also can be seen in games or other emulators, like Dolphin for example, and can be 'fixed' in the same way.

**Comment by internetakias** [comment]
Yeah, the game runs at fullspeed for me, but the audio is lagging behind a bit.

**Comment by Nashismo** [comment]
I am not that stupid to not know about Maximum Performance mode. I tried that and many other things before coming up with this solution. Also I wasn't actually saying audio lag, I din't mean to say that (forgive my english) I meant audio distortion. My game can run 60fps ALL THE TIME and I still get audio distrortion. Please don't assume people are so stupid. 

The only way I was able to fix it was using the SMAA injector (by accident). Same thing used to happen to me on PS1 emulators, is like the emulator fall asleep and there is no Maximum Performance mode to fix this, not even leaving my CPU at 100% all the time. It must be a BIOS problem or an Emulator problem (one or the other). But believe me, I am not the only one with this issue.

**Comment by CellRuby** [comment]
Hi Nashismo, could you please explain how you made the SMAA injector work?
When I run Demul the gpu plugin disappear, I guess I'm putting the files in the wrong place. Yep, I'm a noob at this.

**Comment by Nashismo** [comment]
Man I dont remember exactly now just go to the link I provided and use the
SMAA injector for DX10 (not DX9). You just put the files on the same folder
where your executable, in this case Demul, to work.

I don't have my PC anymore, I have an old laptop and doing other stuff.
Cheers, and believe me SMAA will fix your problems.

**Comment by RinMaru** [comment]
@p1pkin Hey hope your doing alright ^^ Love Demul Ive ditched all the other emus i hope to see the next build soon
